### DISCUSSION TITLE
Doc clarification: conditionals

### DISCUSSION THREAD

**Discussion by candlerb** [keycloak/keycloak#10494] [Ideas]
The documentation [here](https://www.keycloak.org/docs/latest/server_admin/index.html#conditional) for conditionals in authentication flows says:

> * If all executions evaluate as true, the Conditional sub-flow acts as Required.
> * If all executions evaluate as false, the Conditional sub-flow acts as Disabled.

However, it doesn't make it clear what happens if *some* executions evaluate as false.

Does the following more accurately described the intended behaviour?

> * If all executions evaluate as true, the Conditional sub-flow acts as Required.
> * If **any** execution evaluates as false, the Conditional sub-flow acts as Disabled.

Also: if this is the case, does evaluation stop at the first execution which returns false?  If so, maybe it could be clarified as:

> * If all executions evaluate as true, the Conditional sub-flow acts as Required.
> * If **any** execution evaluates as false, then the Conditional sub-flow acts as Disabled. Any remaining executions within the Conditional sub-flow are skipped.

One other thing I noticed [here](https://www.keycloak.org/docs/latest/server_admin/index.html#_detect_existing_user_first_loging_flow):

> Set the authenticator requirement to `Mandatory`

(this occurs twice).  I think it should say `Required` rather than `Mandatory`

**Comment by slominskir** [comment]
For all vs any see: https://github.com/keycloak/keycloak-documentation/pull/1773

**Comment by slominskir** [comment]
For MANDATORY vs REQUIRED see: https://github.com/keycloak/keycloak-documentation/pull/1775

**Comment by mposolda** [comment]
Merged both suggestions https://github.com/keycloak/keycloak-documentation/pull/1773 and https://github.com/keycloak/keycloak-documentation/pull/1775
